provider/cmd/pulumi-resource-nuage/nuage_provider/container_function.py

```diff
@@ -39,162 +46,253 @@ def docker_value(self) -> str:
-                extra_options=["--platform", architecture.docker_value, "--quiet"]
+                extra_options=extra_options,
             )
             image_ignore_changes = []
         else:
+            tmp = tempfile.NamedTemporaryFile(dir="./", delete=True)
             # Use default aws lambda docker image.
-            with open('Dockerfile.awslambda.generated','w') as f:
-                f.writelines([
-                    'FROM public.ecr.aws/lambda/provided:al2',
-                    '\n'
-                    'CMD [ "function.handler" ]'
-                ])
+            with open(tmp.name, "w") as f:
+                f.writelines(
+                    [
+                        "FROM public.ecr.aws/lambda/provided:al2",
+                        "\n" 'CMD [ "function.handler" ]',
+                    ]
+                )
 
             build = docker.DockerBuild(
                 context="./",
-                dockerfile="./Dockerfile.awslambda.generated",
-                extra_options=["--platform", architecture.docker_value, "--quiet"]
+                dockerfile=tmp.name,
+                extra_options=extra_options,
             )
             # Ignore changes on image_name if default docker image is used.
             image_ignore_changes = ["image_name"]
-        
+
         # Authenticate with ECR and get cridentals.
-        registry_id = repository.apply(lambda x: x.split(".")[0])
+        registry_id = repository.repository_url.apply(lambda x: x.split(".")[0])
         auth = aws.ecr.get_authorization_token(registry_id=registry_id)
-        
+
         # Build and Push docker Image.
         image = docker.Image(
-            name=f"{name}-image",
-            build = build,
-            image_name = repository,
+            name=f"{resource_name}-image",
+            build=build,
+            image_name=repository.repository_url.apply(
+                lambda url: f"{url}:{resource_name}"
+            ),
+            local_image_name=f"{pulumi.get_organization()}:{resource_name}",
             registry=docker.ImageRegistry(
                 server=auth.proxy_endpoint,
                 username=auth.user_name,
-                password=auth.password
+                password=auth.password,
             ),
             opts=pulumi.ResourceOptions(
-                parent=self,
-                ignore_changes=image_ignore_changes
+                parent=self, ignore_changes=image_ignore_changes
+            ),
+        )
+
+        # Untag ecs urls and keep only {pulumi.get_organization()}:{resource_name} one.
+        image.image_name.apply(
+            lambda image_name: (
+                local.Command(
+                    f"untag-{resource_name}-image",
+                    create=pulumi.Output.all(
+                        repository.repository_url, resource_name
+                    ).apply(
+                        lambda args: f"docker rmi {args[0]}:{args[1]} && docker rmi {image_name}"
+                    ),
```

Comment:
Does this actually work ?

Shouldn't it be `image_name` instead of `resource_name` ? 
We want to untag the automated naming done by Pulumi, right ?

I also suggest you switch to kwargs-style for pulumi.Output.all as it is easier to read :)